tests/test_rapids_dep_file_generator.py

```diff
@@ -0,0 +1,59 @@
+from unittest import mock
+from rapids_dep_file_generator.rapids_dep_file_generator import (
+    dedupe,
+    make_dependency_file,
+)
+from rapids_dep_file_generator.constants import cli_name
+import yaml
+
+
+def test_dedupe():
+    # simple list
+    deduped = dedupe(["dep1", "dep1", "dep2"])
+    assert deduped == ["dep1", "dep2"]
+
+    # list w/ pip dependencies
+    deduped = dedupe(
+        [
+            "dep1",
+            "dep1",
+            {"pip": ["pip_dep1", "pip_dep2"]},
+            {"pip": ["pip_dep1", "pip_dep2"]},
+        ]
+    )
+    assert deduped == ["dep1", {"pip": ["pip_dep1", "pip_dep2"]}]
+
+
+@mock.patch("rapids_dep_file_generator.rapids_dep_file_generator.relpath")
+def test_make_dependency_file(mock_relpath):
+    relpath = "../../dependencies.yaml"
```

Comment:
This overrides the imported name `from os.path import relpath`. I was looking for a reason to say "use `import os.path` and call `os.path.join` instead of `from os.path import join`" and this is the kind of thing that justifies that statement. 😄